File: Java/PowerSupply.java
public class PowerSupply extends Komponen {
    private int wattage;
    private String efficiency; // 80+ Bronze, Gold, etc
    private boolean modular;

    public PowerSupply() {
        super();
        this.wattage = 0;
        this.efficiency = "";
        this.modular = false;
    }

    public PowerSupply(int wattage, String efficiency, boolean modular, String merk, String nama, int harga) {
        super(merk, nama, harga);
        this.wattage = wattage;
        this.efficiency = efficiency;
        this.modular = modular;
    }

    public void setWattage(int wattage) {
        this.wattage = wattage;
    }

    public void setEfficiency(String efficiency) {
        this.efficiency = efficiency;
    }

    public void setModular(boolean modular) {
        this.modular = modular;
    }

    public int getWattage() {
        return this.wattage;
    }

    public String getEfficiency() {
        return this.efficiency;
    }

    public boolean isModular() {
        return this.modular;
    }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("Wattage: " + this.wattage + "W");
        System.out.println("Efficiency: " + this.efficiency);
        System.out.println("Modular: " + (this.modular ? "Ya" : "Tidak"));
    }
}

File: Cpp/PowerSupply.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
using namespace std;

class PowerSupply : public Komponen {
private:
    int wattage;
    string efficiency;  // 80+ Bronze, Gold, etc
    bool modular;

public:
    PowerSupply() : Komponen() {
        this->wattage = 0;
        this->efficiency = "";
        this->modular = false;
    }

    PowerSupply(int wattage, string efficiency, bool modular, string merk, string nama, int harga) 
        : Komponen(merk, nama, harga) {
        this->wattage = wattage;
        this->efficiency = efficiency;
        this->modular = modular;
    }

    void setWattage(int wattage) {
        this->wattage = wattage;
    }

    void setEfficiency(string efficiency) {
        this->efficiency = efficiency;
    }

    void setModular(bool modular) {
        this->modular = modular;
    }

    int getWattage() const {
        return this->wattage;
    }

    string getEfficiency() const {
        return this->efficiency;
    }

    bool isModular() const {
        return this->modular;
    }

    void displayInfo() const override {
        Komponen::displayInfo();
        cout << "Wattage: " << this->wattage << "W" << endl;
        cout << "Efficiency: " << this->efficiency << endl;
        cout << "Modular: " << (this->modular ? "Ya" : "Tidak") << endl;
    }

    ~PowerSupply() {}
};
